elm/elm.json

```diff
@@ -6,20 +6,25 @@
     "elm-version": "0.19.1",
     "dependencies": {
         "direct": {
+            "elm/browser": "1.0.2",
             "elm/core": "1.0.5",
-            "elm/json": "1.1.3",
-            "elm/random": "1.0.0",
-            "elm/time": "1.0.0"
+            "elm/html": "1.0.0"
         },
-        "indirect": {}
+        "indirect": {
+            "elm/json": "1.1.3",
+            "elm/time": "1.0.0",
+            "elm/url": "1.0.0",
+            "elm/virtual-dom": "1.0.3"
+        }
     },
     "test-dependencies": {
         "direct": {
-            "elm-explorations/test": "1.2.0"
+            "elm-explorations/test": "2.0.0"
         },
         "indirect": {
-            "elm/html": "1.0.0",
-            "elm/virtual-dom": "1.0.2"
+            "elm/bytes": "1.0.8",
+            "elm/random": "1.0.0",
+            "turboMaCk/queue": "1.1.0"
```

Comment:
Are all these changes intentional?

- elm-review flags for example `elm/browser` as unused.
- Does it work to move some depenencies from direct to indirect like that?

Reply:
I am now very confused what this `elm/elm.json` file is for.

When running tests, we generate an Elm application deep in `elm-stuff/` with source-directories that include:

- Some generated Elm code in `elm-stuff/`.
- The `elm/` folder in this repo (it is part of the npm package).
- Folders from the user’s project.

`elm/elm.json` is explicitly included in the npm package: https://github.com/rtfeldman/node-test-runner/blob/026581eb144a32defd6dae3c3de5c46e5e987527/package.json#L32. But I wonder if that is a mistake.

My current thinking is that `elm/elm.json` only exists to:

1. Make it easier to work on the Elm code in `elm/` – with an elm.json editor integrations understand the code.
2. Run the unit tests of `elm/`.

But when actually running `elm-test`, I think we add the same dependencies as in `elm/elm.json` in this solver code: https://github.com/rtfeldman/node-test-runner/blob/026581eb144a32defd6dae3c3de5c46e5e987527/lib/Solve.js#L61-L64.

I think we should:

- Check if my understanding of `elm/elm.json` is correct, and if so continue with the following points.
- I guess it doesn’t matter super much which dependencies are in `elm/elm.json` as long as elm-review and the tests are happy.
- Make sure that the solver code adds all the necessary dependencies for 2.0. Is `turboMaCk/queue` needed now? I’m a bit confused.
- Add a code comment in the solver code explaining what needs to be in sync.
- Remove `elm/elm.json` from package.json.

Reply:
`turboMaCk/queue` is needed for elm-explorations/test master / future 2.0.0

Reply:
After testing, my understanding seems correct. I removed `node_modules/elm-test/elm/elm.json` in a project and it still worked. So it’s not needed.

I went ahead and cleaned up `elm/elm.json`, removed it from `package.json` and added comments in the solver code.